packages/@uppy/core/types/index.d.ts

```diff
@@ -21,7 +21,7 @@ export type FileProgress = UppyUtils.FileProgress;
 export type FileRemoveReason = 'removed-by-user' | 'cancel-all';
 
 // Replace the `meta` property type with one that allows omitting internal metadata addFile() will add that
-type UppyFileWithoutMeta<TMeta, TBody> = OmitKey<
+type UppyFileWithoutMeta<TMeta extends IndexedObject<any>, TBody extends IndexedObject<any>> = OmitKey<
```

Comment:
Can we avoid a lot of duplication here by creating a new type that already extends it?

Reply:
IIUC here we are defining the type `TMeta`, so I don't think that's an option.

Reply:
If `TMeta` is always with `extends IndexedObject<any>`, it can become the new type. If there are still exceptions, we can create a new type with combines the two and use those where needed in this file.

Reply:
I'm defining `TMeta` here, I don't think I can "make it a new type", or I don't understand what you mean.